### PR TITLE
Refactor createRainbowSprinkles to serialize function

### PR DESCRIPTION
**Pull request by roginfarrer**
## Description

Discovered a major issue where we were bundling `@vanilla-extract/css` into the final bundle within apps. This was due to the fact that we documented running `createRainbowSprinkles` within a non `.css.js` file, which was required since the function returned non-serialized functions. However now that I understand a bit more about serialization (and vanilla-extract's utils for it), it allowed some substantial improvements to the API. So this was a somewhat substantial rewrite.

The breaking change is to the initial set-up of rainbow sprinkles, but doesn't impact the use of components that consume rainbow sprinkles.

Before:

```js
// rainbow-sprinkles.js
export const {
  getBoxProps, 
  extractSprinklesFromProps, 
  createRainbowSprinklesCss
} = createRainbowSprinkles({ ... })
```

After:

```js
// rainbow-sprinkles.css.js
export const rainbowSprinkles = createRainbowSprinkles({ ... })
```

So now `createRainbowSprinkles` can be called within a `.css.js` file.

## Type of Change

- [ ] Bug Fix
- [ ] New Feature
- [x] Breaking Change
- [x] Refactor
- [x] Documentation
- [ ] Other (please describe)

## Checklist

- [x] I have read the [contributing guidelines](https://github.com/wayfair-incubator/oss-template/blob/main/CONTRIBUTING.md)
- [x] Existing issues have been referenced (where applicable)
- [x] I have verified this change is not present in other open pull requests
- [x] Functionality is documented
- [x] All code style checks pass
- [x] New code contribution is covered by automated tests
- [x] All new and existing tests pass

### CODE REVIEW

File: packages/rainbow-sprinkles/src/__tests__/createRainbowSprinkles.test.ts [hunk 1]
```diff
@@ -30,373 +30,273 @@ const vars = {
...
+      paddingX: ['paddingLeft', 'paddingRight'],
+      paddingY: ['paddingTop', 'paddingBottom'],
+      px: ['paddingLeft', 'paddingRight'],
+      py: ['paddingTop', 'paddingBottom'],
+      bg: ['background'],
+      placeItems: ['alignItems', 'justifyContent'],
+    },
   });
 
   test('getBoxProps', () => {
```
Comment (hamlim) [inline packages/rainbow-sprinkles/src/__tests__/createRainbowSprinkles.test.ts:75]: Thoughts on renaming this and other describe blocks from `getBoxProps` to `rainbowSprinkles`?

File: packages/rainbow-sprinkles/src/assignClasses.ts [hunk 2]
```diff
@@ -29,7 +26,7 @@ export function assignClasses<Conditions extends BaseConditions>(
     // Quietly warn
     // eslint-disable-next-line no-console
     console.error(
-      `Homebase: invalid value provided to prop '${propName}'. Expected one of ${Object.keys(
+      `Rainbow Sprinkles: invalid value provided to prop '${propName}'. Expected one of ${Object.keys(
```
Comment (hamlim) [inline packages/rainbow-sprinkles/src/assignClasses.ts:29]: Heh noticed this today also, I'll remove my change in #25 👍

This will resolve #22 also 💯

File: example/components/Box.tsx [hunk 3]
```diff
@@ -16,14 +11,11 @@ export const Box = <C extends ElementType = 'div'>({
   children,
   ...props
 }: BoxProps<C>) => {
-  const { sprinkles, otherProps } = extractSprinklesFromProps(props);
   const Component = as || 'div';
+  const { className, style, otherProps } = rainbowSprinkles(props);
```
[the remark this answers is not in the document]
Comment (hamlim) [reply example/components/Box.tsx:15]: Ok, replying to this after finishing reading the diff. Seems like `rainbowSprinkles` will effectively ignore `props.style` and `props.className`, which I think makes sense - can leave the merging for libraries implementing rainbow-sprinkles to let them determine precedence when merging if they want to merge at all.

tl;dr: Implementation seems fine as-is

**Review by hamlim — APPROVED** [review]
This seems pretty dope :shipit:

File: packages/rainbow-sprinkles/README.md [hunk 4]
```diff
@@ -17,6 +17,22 @@ Compared to sprinkles:
```
[the remark this answers is not in the document]
Comment (roginfarrer) [reply packages/rainbow-sprinkles/README.md:20]: Good catch, I removed it 👍🏻

**Review by hamlim — APPROVED** [review]
